File: aidp-python-client/src/aidp_python_client/aidataplatform_dp/models/execute_sql_command_details.py
# coding: utf-8
# Copyright (c) 2016, 2026, Oracle and/or its affiliates.  All rights reserved.
# This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.

from .execute_database_user_workflows_details import ExecuteDatabaseUserWorkflowsDetails
from oci.util import formatted_flat_dict, NONE_SENTINEL, value_allowed_none_or_none_sentinel  # noqa: F401
from oci.decorators import init_model_state_from_kwargs


@init_model_state_from_kwargs
class ExecuteSqlCommandDetails(ExecuteDatabaseUserWorkflowsDetails):
    """
    Running SQL command.
    """

    def __init__(self, **kwargs):
        """
        Initializes a new ExecuteSqlCommandDetails object with values from keyword arguments. The default value of the :py:attr:`~oci.aidataplatform_dp.models.ExecuteSqlCommandDetails.action_type` attribute
        of this class is ``EXECUTE_SQL_COMMAND`` and it should not be changed.
        The following keyword arguments are supported (corresponding to the getters/setters of this class):

        :param action_type:
            The value to assign to the action_type property of this ExecuteSqlCommandDetails.
        :type action_type: str

        :param user_schema_name:
            The value to assign to the user_schema_name property of this ExecuteSqlCommandDetails.
        :type user_schema_name: str

        :param sql_command:
            The value to assign to the sql_command property of this ExecuteSqlCommandDetails.
        :type sql_command: str

        """
        self.swagger_types = {
            'action_type': 'str',
            'user_schema_name': 'str',
            'sql_command': 'str'
        }

        self.attribute_map = {
            'action_type': 'actionType',
            'user_schema_name': 'userSchemaName',
            'sql_command': 'sqlCommand'
        }

        self._action_type = None
        self._user_schema_name = None
        self._sql_command = None
        self._action_type = 'EXECUTE_SQL_COMMAND'

    @property
    def user_schema_name(self):
        """
        Gets the user_schema_name of this ExecuteSqlCommandDetails.
        The user schema name. Only supports a valid knowledge base user and ADMIN.


        :return: The user_schema_name of this ExecuteSqlCommandDetails.
        :rtype: str
        """
        return self._user_schema_name

    @user_schema_name.setter
    def user_schema_name(self, user_schema_name):
        """
        Sets the user_schema_name of this ExecuteSqlCommandDetails.
        The user schema name. Only supports a valid knowledge base user and ADMIN.


        :param user_schema_name: The user_schema_name of this ExecuteSqlCommandDetails.
        :type: str
        """
        self._user_schema_name = user_schema_name

    @property
    def sql_command(self):
        """
        Gets the sql_command of this ExecuteSqlCommandDetails.
        Base64-encoded SQL command.


        :return: The sql_command of this ExecuteSqlCommandDetails.
        :rtype: str
        """
        return self._sql_command

    @sql_command.setter
    def sql_command(self, sql_command):
        """
        Sets the sql_command of this ExecuteSqlCommandDetails.
        Base64-encoded SQL command.


        :param sql_command: The sql_command of this ExecuteSqlCommandDetails.
        :type: str
        """
        self._sql_command = sql_command

    def __repr__(self):
        return formatted_flat_dict(self)

    def __eq__(self, other):
        if other is None:
            return False

        return self.__dict__ == other.__dict__

    def __ne__(self, other):
        return not self == other

File: aidp-java-client/src/main/java/com/oracle/aidataplatform/dp/model/ExecuteSqlCommandDetails.java
// Copyright (c) 2026, Oracle and/or its affiliates.  All rights reserved.

package com.oracle.aidataplatform.dp.model;



/**
 * Running SQL command.
**/
@jakarta.annotation.Generated(value = "OracleSDKGenerator", comments = "API Version: 20260430")
@com.fasterxml.jackson.databind.annotation.JsonDeserialize(builder=ExecuteSqlCommandDetails.Builder.class)
@com.fasterxml.jackson.annotation.JsonTypeInfo(use=com.fasterxml.jackson.annotation.JsonTypeInfo.Id.NAME, include=com.fasterxml.jackson.annotation.JsonTypeInfo.As.PROPERTY, property="actionType")

public final class ExecuteSqlCommandDetails extends ExecuteDatabaseUserWorkflowsDetails {
    @com.fasterxml.jackson.databind.annotation.JsonPOJOBuilder(withPrefix = "")
    public static class Builder {
                /**
     * The user schema name. Only supports a valid knowledge base user and ADMIN.
     **/
    
@com.fasterxml.jackson.annotation.JsonProperty("userSchemaName")
private String userSchemaName;

        /**
         * The user schema name. Only supports a valid knowledge base user and ADMIN.
         * @param userSchemaName the value to set
         * @return this builder
         **/
        

public Builder userSchemaName(String userSchemaName) {
    this.userSchemaName = userSchemaName;
    return this;
}
            /**
     * Base64-encoded SQL command.
     **/
    
@com.fasterxml.jackson.annotation.JsonProperty("sqlCommand")
private String sqlCommand;

        /**
         * Base64-encoded SQL command.
         * @param sqlCommand the value to set
         * @return this builder
         **/
        

public Builder sqlCommand(String sqlCommand) {
    this.sqlCommand = sqlCommand;
    return this;
}


        public ExecuteSqlCommandDetails build() {
            ExecuteSqlCommandDetails model = new ExecuteSqlCommandDetails(this.userSchemaName
                , this.sqlCommand);            return model;
        }

        @com.fasterxml.jackson.annotation.JsonIgnore
        public Builder copy(ExecuteSqlCommandDetails model) {
                this.userSchemaName(model.getUserSchemaName());
    this.sqlCommand(model.getSqlCommand());
return this;
        }
    }

    /**
     * Create a new builder.
     */
    public static Builder builder() {
        return new Builder();
    }


    public Builder toBuilder() {
        return new Builder().copy(this);
    }

    
    @Deprecated
    public ExecuteSqlCommandDetails(String userSchemaName, String sqlCommand) {
    super();
        this.userSchemaName = userSchemaName;
        this.sqlCommand = sqlCommand;
    }


        /**
     * The user schema name. Only supports a valid knowledge base user and ADMIN.
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("userSchemaName")
    private final String userSchemaName;

        /**
     * The user schema name. Only supports a valid knowledge base user and ADMIN.
     * @return the value
     **/
    
    public String getUserSchemaName() {
        return userSchemaName;
    }


        /**
     * Base64-encoded SQL command.
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("sqlCommand")
    private final String sqlCommand;

        /**
     * Base64-encoded SQL command.
     * @return the value
     **/
    
    public String getSqlCommand() {
        return sqlCommand;
    }

    @Override
    public String toString() {
        return this.toString(true);
    }

    /**
     * Return a string representation of the object.
     * @param includeByteArrayContents true to include the full contents of byte arrays
     * @return string representation
     */
    public String toString(boolean includeByteArrayContents) {
        java.lang.StringBuilder sb = new java.lang.StringBuilder();
        sb.append("ExecuteSqlCommandDetails(");
        sb.append("super=").append(super.toString(includeByteArrayContents));
        sb.append(", userSchemaName=").append(String.valueOf(this.userSchemaName));
        sb.append(", sqlCommand=").append(String.valueOf(this.sqlCommand));
        sb.append(")");
        return sb.toString();
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) {
            return true;
        }
        if (!(o instanceof ExecuteSqlCommandDetails)) {
            return false;
        }

        ExecuteSqlCommandDetails other = (ExecuteSqlCommandDetails) o;
        return java.util.Objects.equals(this.userSchemaName, other.userSchemaName) &&
            java.util.Objects.equals(this.sqlCommand, other.sqlCommand) &&
            super.equals(other);
    }

    @Override
    public int hashCode() {
        final int PRIME = 59;
        int result = super.hashCode();
        result = (result * PRIME) + (this.userSchemaName == null ? 43 : this.userSchemaName.hashCode());
        result = (result * PRIME) + (this.sqlCommand == null ? 43 : this.sqlCommand.hashCode());
        return result;
    }


}
